features/events/traps/src/main/java/org/opennms/netmgt/trapd/EventCreator.java

```diff
@@ -41,139 +41,86 @@
 import org.opennms.netmgt.snmp.SnmpValue;
+import org.opennms.netmgt.snmp.SnmpVarBindDTO;
 import org.opennms.netmgt.snmp.TrapIdentity;
-import org.opennms.netmgt.snmp.TrapProcessor;
+import org.opennms.netmgt.snmp.TrapInformation;
 import org.opennms.netmgt.xml.event.Event;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * <p>EventCreator class.</p>
- *
- * @author ranger
- * @version $Id: $
- */
-public class EventCreator implements TrapProcessor {
+import com.google.common.base.Strings;
+
+class EventCreator {
 	
 	private static final Logger LOG = LoggerFactory.getLogger(EventCreator.class);
-    
-    private final EventBuilder m_eventBuilder;
 
     private final InterfaceToNodeCache m_cache;
 
-    private String m_systemId;
-
-    private String m_location;
-
-    private InetAddress m_trapAddress;
-
     public EventCreator(InterfaceToNodeCache cache) {
-        m_cache = cache;
-        m_eventBuilder = new EventBuilder(null, "trapd");
-    }
-
-    @Override
-    public void setSystemId(String systemId) {
-        m_systemId = systemId;
-    }
-
-    @Override
-    public void setLocation(String location) {
-        m_location = location;
+        m_cache = Objects.requireNonNull(cache);
     }
 
-    /** {@inheritDoc} */
-    @Override
-    public void setCommunity(String community) {
-        m_eventBuilder.setCommunity(community);
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void setCreationTime(long creationTime) {
-        m_eventBuilder.setTime(new Date(creationTime));
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void setTimeStamp(long timeStamp) {
-        m_eventBuilder.setSnmpTimeStamp(timeStamp);
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void setVersion(String version) {
-        m_eventBuilder.setSnmpVersion(version);
-    }
-
-    private void setGeneric(int generic) {
-        m_eventBuilder.setGeneric(generic);
-    }
-
-    private void setSpecific(int specific) {
-        m_eventBuilder.setSpecific(specific);
-    }
-
-    private void setEnterpriseId(String enterpriseId) {
-        m_eventBuilder.setEnterpriseId(enterpriseId);
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void setAgentAddress(InetAddress agentAddress) {
-        m_eventBuilder.setHost(InetAddressUtils.toIpAddrString(agentAddress));
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void processVarBind(SnmpObjId name, SnmpValue value) {
-        m_eventBuilder.addParam(SyntaxToEvent.processSyntax(name.toString(), value));
-        if (EventConstants.OID_SNMP_IFINDEX.isPrefixOf(name)) {
-            m_eventBuilder.setIfIndex(value.toInt());
+    public Event getEvent(TrapInformation trapInformation) {
+        // Validate trap before continuing
+        trapInformation.validate();
+
+        // TODO MVR make log message nice
+        LOG.debug("{} trap - trapInterface: ()", trapInformation.getVersion(), trapInformation.getTrapAddress());
```

Comment:
At the moment we do not have a consistend way of addressing this. Sometimes we just log, sometimes we use `isLogLevelEnabled`. If the logging is not very heavy, I would let Java and the logging framework taking care of optimizations.